components/event-publisher-proxy/pkg/handler/handler_v1alpha2_test.go

```diff
@@ -322,6 +330,210 @@ func TestHandler_publishCloudEventsV1Alpha2(t *testing.T) {
 	}
 }
 
+func TestHandler_publishLegacyEventsAsCEV1alpha2(t *testing.T) {
+	type fields struct {
+		Sender            sender.GenericSender
+		LegacyTransformer legacy.RequestToCETransformer
+		collector         metrics.PublishingMetricsCollector
+		eventTypeCleaner  eventtype.Cleaner
+	}
+	type args struct {
+		request *http.Request
+	}
+
+	const bucketsFunc = "Buckets"
+	latency := new(mocks.BucketsProvider)
+	latency.On(bucketsFunc).Return(nil)
```

Comment:
This mock is used by metrics collector. The metrics collector is tested at the end of each test case (`metricstest.EnsureMetricMatchesTextExpositionFormat(t, h.collector, tt.wantTEF)`), so this would be covered. 

Also, latency mock is defined globally for all test cases, so doing `latency.AssertExpectations(t)` would be green even if its called by any single test case. To test it per test case, we have to define a new latency mock per test case and check `latency.AssertExpectations(t)`.

Reply:
Discussed in person.